### ISSUE TITLE
Model does not load in the expected dtype

### ISSUE THREAD

**Issue by AugustasMacijauskas** [CarperAI/trlx#535]
### 🐛 Describe the bug

I am using `trlx` to train on my own dataset. I am using [this `trlx` example](https://github.com/CarperAI/trlx/blob/main/examples/hh/ppo_hh.py) as the starter code.

The related code change that I make is the following in the `default_config`:
```python
- model=ModelConfig(model_path="EleutherAI/gpt-j-6B", num_layers_unfrozen=2)
+ model=ModelConfig(model_path="gpt2-xl")
```

I also use the `accelerate` config [provided in trlx](https://github.com/CarperAI/trlx/blob/main/configs/accelerate/zero2-bf16.yaml), and launch training using the same command as in the corresponding README.

I expect the weights to be loaded in `bfloat16`, but if I add a print statement to print the model weight `dtype` after [this line](https://github.com/CarperAI/trlx/blob/0c94ee88598c549b0172d685c987fdde054b75a4/trlx/trainer/accelerate_ppo_trainer.py#L66), it says `float32`. The print statement that I add is:
```python
print(f"{self.accelerator.unwrap_model(self.model).base_model.transformer.h[47].mlp.c_fc.weight.dtype=}")
```

This later causes an OOM error which I think would be resolved if I were able to load the model in `bfloat16`.

My GPU is `NVIDIA A100-SXM4-40GB` which supports `bfloat16`.

### Which trlX version are you using?

`trlx==0.7.0`

### Additional system and package information

Python 3.10.11, transformers 4.28.1, Linux OS

**Comment by maxreciprocate** [comment]
Hi! Seems like I can't reproduce this behaviour. Running using this command on the same `NVIDIA A100-SXM4-40GB`
```bash
accelerate launch --config_file configs/accelerate/zero2-bf16.yaml --num_processes 1 examples/ppo_sentiments.py
```
with modified[1] print statement would give me:

```self.accelerator.unwrap_model(self.model).base_model.transformer.h[0].mlp.c_fc.weight.dtype=torch.bfloat16```

[1] you mentioned that you've changed the default config to use `gpt2-xl`, but it doesn't have 48 layers for the print statement you've added to not crash with IndexError

**Comment by AugustasMacijauskas** [comment]
@maxreciprocate Okay, this is really confusing, I've tried running my code again and it still prints `float32`. The only thing that I had different and forgot to mention previously is that I am running on a cluster with 8 GPUs, but I don't think it should make a difference since I've now tried running on one GPU and it still fails.

I'll try checking if any other changes to the code could be the reason this happens.
